soc/arm/atmel_sam/common/soc_gpio.h

```diff
@@ -121,7 +136,11 @@ void soc_gpio_list_configure(const struct soc_gpio_pin pins[],
  */
 static inline void soc_gpio_set(const struct soc_gpio_pin *pin)
 {
+#ifdef ID_PIOA
 	pin->regs->PIO_SODR = pin->mask;
+#else
+	pin->regs->OVRS = pin->mask;
+#endif
```

Comment:
Some small package devices with limited number of pins may have e.g. PIOB and PIOC but not PIOA module. Looking at the SAM SoCs currently supported the above approach should work but it may be safer to test for `PIO_SODR_OFFSET` macro to verify existence of the register. The same holds true for the remaining registers.

Reply:
Good point! I checked and `PIO_SODR_OFFSET` doesn't exists for SAM4S/E. This means that `PIO_SODR_OFFSET` can't be used. I searched on ASF and noted that ID_GPIO is only defined for SAM4L.
```
#ifdef ID_GPIO
	Gpio *regs;        /** pointer to registers of the GPIO controller */
#else
	Pio *regs;         /** pointer to registers of the PIO controller */
#endif
```